### PR TITLE
[WIP] v0.3 nitpicks and tweaks

### PR DESCRIPTION
**Pull request by betatim**
Minor fixes and nitpicks for preparing v0.3 release.

Adds auto deployment directly from travis to pypi when we tag a new release.

Fixing up a few doc strings.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -25,3 +25,16 @@ after_success:
...
     fi
+
+deploy:
+  provider: pypi
+  user: betatim
+  # this line together with `setup.cfg` creates universal wheels as long as
+  # skopt is a python only module (no compiled code)
+  distributions: sdist bdist_wheel
+  on:
+    tags: true
```
Comment (MechCoder) [inline .travis.yml]: So does this deploy when I push a tag?

**Comment by MechCoder** [comment]
When you are at it, I think we can automate github releases as well (https://docs.travis-ci.com/user/deployment/releases/)

File: .travis.yml (hunk 1 above)
Comment (betatim) [inline .travis.yml]: I think so. It will definitely deploy when you create a new release (which is a tag?!) on github.

**Comment by betatim** [comment]
> When you are at it, I think we can automate github releases as well (https://docs.travis-ci.com/user/deployment/releases/)

for python packages does anyone even want to download stuff from the github release page? Had briefly considered it but then decided that pypi is really the only thing we care about. At least I can't remember the last time I went to a github repo to download stuff for a python package.

One day we should create a conda-forge package though.

**Comment by MechCoder** [comment]
Thanks!

**Pull request merged**